### PR TITLE
connect to Api and implement the add and remove functionality

### PR DESCRIPTION
**Pull request by naji-97**
- [ ] Use [axios](https://axios-http.com/docs/intro) with [createAsyncThunk](https://redux-toolkit.js.org/api/createAsyncThunk) to make the HTTP requests to the backend
- [ ] Fetch the books from the Bookstore API after the component has rendered
- [ ] Refactor the add book and remove book features to persist your changes in the server

### CODE REVIEW

**Review by DammyShittu — COMMENTED** [review]
Hi @naji-97,

While you made a great effort in this project, I cannot proceed to review your code.

## Invalid Code Review Request

### You have submitted a project that has no linter checks. Kindly set up your linter checks for this PR.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: src/components/BookList.js [hunk 1]
```diff
@@ -1,17 +1,23 @@
...
 import Form from './Form';
+import { fetchBooks } from '../redux/books/booksSlice';
 
 function BookList() {
+  const dispatch = useDispatch();
+
   const books = useSelector((state) => state.books.books);
+  useEffect(() => {
+    dispatch(fetchBooks());
+  }, [dispatch]);
```
Comment (Zuhaib042) [inline src/components/BookList.js:13]: - [ ] Good job but there is a bug in your project that you need to fix before proceeding. When I add new books in your project, and then if I try to remove any of them, they all get disappeared and then I have to refresh the page to fetch them. it happens specifically whenever there are no books and then you add any number and then the first time you want to remove. After refreshing and getting all the books back, the removing functionality works fine.
here is the video

https://github.com/naji-97/sec-book-store/assets/96683843/1c26b618-0ab7-4b53-9ec5-d550ee45f43b

**Review by Zuhaib042 — CHANGES_REQUESTED** [review]
Hi @naji-97 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
✅ Linters passing
✅ Descriptive PR

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by OyePriscilla — APPROVED** [review]
Hi @naji-97 👏 👏, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
Implement functionality to remove book ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**